plugins/inputs/aerospike/aerospike.go

```diff
@@ -65,67 +72,231 @@ func (a *Aerospike) gatherServer(hostport string, acc telegraf.Accumulator) erro
 		iport = 3000
 	}
 
-	c, err := as.NewClient(host, iport)
+	var c *as.Client
+	if urlInfo.User != nil {
+		userInfo := urlInfo.User
+		clientPolicy := as.NewClientPolicy()
+		clientPolicy.User = userInfo.Username()
+		clientPolicy.Password, _ = userInfo.Password()
```

Comment:
could you add TLS support here as well? see https://github.com/influxdata/telegraf/blob/master/internal/internal.go#L118 and see the influxdb output plugin sample config for how to format the config options.